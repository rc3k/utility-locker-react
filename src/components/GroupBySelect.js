import React from 'react';

export default ({setGroupBy, field}) => (
  <div className="d-flex w-100 form-group p-2" >
    <label htmlFor="group-by-select" className="pt-2 pr-1 text-right">
      Group by:
    </label>
    <select
      id="group-by-select"
      className="form-control"
      value={field}
      onChange={event => setGroupBy(event.target.value)}
    >
      <option value="">None</option>
      <option value="type">Type</option>
    </select>
  </div>
);
