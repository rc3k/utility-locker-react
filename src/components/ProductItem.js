import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions/item';

export class ProductItem extends React.Component {
  componentDidMount() {
    const { loadItem, match } = this.props;
    const { itemid } = match.params;
    if (itemid) {
      loadItem('products', itemid);
    }
  }

  render() {
    const { fields, meta, history } = this.props;
    if (meta.status === 'REQUESTED') {
      return (
        <div className="loader">Loading...</div>
      )
    }
    return (
      <div className="item">
        <h1>{fields.name}</h1>
        <p>A stub for an item view</p>
        <input type="button" onClick={() => history.push('/')} value="Go Back" />
      </div>
    );
  }
}

export const ProductItemContainer = withRouter(
  connect(
    (state) => ({
      fields: state.item.fields,
      meta: state.item.meta,
    }),
    actions,
  )(ProductItem),
);
