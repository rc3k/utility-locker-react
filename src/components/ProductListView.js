import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions/collection';

export class ProductListView extends React.Component {
  componentDidMount() {
    const { loadCollection } = this.props;
    loadCollection('products');
  }

  render() {
    const { items, meta } = this.props;

    if (meta.status === 'REQUESTED') {
      return (
        <div className="loader">Loading...</div>
      )
    }

    return (
      <div className="collection">
        <h1>{meta.collectionName}</h1>
        <table>
          <thead>
            <tr>
              <th>Name</th>
            </tr>
          </thead>
          <tbody>
            {items.map((item) => (
              <tr key={item.id}>
                <td>
                  <Link to={`/${item.id}`}>{item.name}</Link>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export const ProductListViewContainer = withRouter(
  connect((state, ownProps) => ({
    items: state.products.items,
    meta: state.products.meta,
  }), actions)(ProductListView),
);
