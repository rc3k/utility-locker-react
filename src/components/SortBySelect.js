import React from 'react';

export default ({setSort, column, direction}) => (
  <div className="d-flex w-100 p-2" >
    <label htmlFor="sort-by-select" className="pt-2 pr-1 text-right">
      Sort:
    </label>
    <select
      id="sort-by-select"
      className="form-control"
      value={`${column}_${direction}`}
      onChange={event => setSort(...event.target.value.split('_'))}
    >
      <option value="name_asc">Default (name)</option>
      <option value="price_desc">Highest price</option>
      <option value="price_asc">Lowest price</option>
    </select>
  </div>
);
