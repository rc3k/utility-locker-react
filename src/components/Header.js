import React from 'react';
import SortBySelect from './SortBySelect';
import GroupBySelect from './GroupBySelect';

export default ({
  heading, setSort, sortParams, groupByField, setGroupBy,
}) => (
  <div className="row">
    <div className="col-md-6">
      <h1>{heading}</h1>
    </div>
    <div className="col-md-6">
      <SortBySelect
        setSort={setSort}
        column={sortParams.column}
        direction={sortParams.direction}
      />
      <GroupBySelect
        field={groupByField}
        setGroupBy={setGroupBy}
      />
    </div>
  </div>
);
