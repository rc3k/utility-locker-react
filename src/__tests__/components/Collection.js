import React from 'react';
import renderer from 'react-test-renderer';

import { BrowserRouter as Router } from 'react-router-dom';
import { Products } from '../../components/Products';

it('renders correctly when data is loading', () => {
  const collection = renderer.create(
    <Router>
      <Products
        meta={{
          status: 'REQUESTED',
        }}
        match={{
          params: {
            collection: 'users',
          },
        }}
        loadCollection={() => {
        }}
      />
    </Router>,
  ).toJSON();
  expect(collection).toMatchSnapshot();
});

it('renders correctly when items have loaded', () => {
  const collection = renderer.create(
    <Router>
      <Products
        items={[
          { id: 1, name: 'Item 1' },
          { id: 2, name: 'Item 2' },
          { id: 3, name: 'Item 3' },
        ]}
        meta={{
          status: 'SUCCESS',
        }}
        match={{
          params: {
            collection: 'users',
          },
        }}
        loadCollection={() => {
        }}
      />
    </Router>,
  ).toJSON();
  expect(collection).toMatchSnapshot();
});
