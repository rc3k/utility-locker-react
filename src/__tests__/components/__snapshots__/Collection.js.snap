// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when data is loading 1`] = `
<div
  className="loader"
>
  Loading...
</div>
`;

exports[`renders correctly when items have loaded 1`] = `
<div
  className="collection"
>
  <h1 />
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/undefined/1"
            onClick={[Function]}
          >
            Item 1
          </a>
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/undefined/2"
            onClick={[Function]}
          >
            Item 2
          </a>
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/undefined/3"
            onClick={[Function]}
          >
            Item 3
          </a>
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/undefined/item/add"
    onClick={[Function]}
  >
    Add item
  </a>
</div>
`;
